tests/beta/queuing/test_beta_measure.py

```diff
@@ -88,20 +102,24 @@ def test_annotating_tensor_hermitian(self, stat_func, return_type):
     def test_annotating_tensor_return_type(self, op1, op2, stat_func, return_type):
         """Test that the return_type related info is updated for a measurement
         when called for an Tensor observable"""
-        with qml._queuing.AnnotatedQueue() as q:
-            A = op1(0)
-            B = op2(1)
-            tensor_op = BetaTensor(A, B)
-            stat_func(tensor_op)
+        with contextlib.ExitStack() as stack:
+            for mock in mock_operations():
+                stack.enter_context(mock)
```

Comment:
Oh, this is quite cool! So, `contextlib.ExitStack()` provides a method of entering and exiting numerous context managers at once; think of it like a context manager collection.

So instead of doing
```python
with mock1:
    with mock2:
        with mock3:
```
etc., you can instead do
```python
with contextlib.ExitStack() as stack:
    for mock in [mock1, mock2, mock3]:
        stack.enter_context(mock)
```

when you exit the `ExitStack` context, it will automate the process of exiting all contexts in the stack for you.

Here, I'm just using this to easily enter all the mock operation contexts (from `beta/queuing/operations.py`), so that the test can do `A@B` and have it correctly queue itself to the new `AnnotatedQueue`. Essentially, manually doing what the tape does in `__enter__`.